tutorials/manage-instances-with-terraform-and-functions/index.mdx

```diff
@@ -0,0 +1,311 @@
+).
+
+
+<Message type="requirement">
+
+  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
+  - You have [generated an API key](/console/my-project/how-to/generate-api-key/)
+  - You have [Python](https://www.python.org/) installed on your machine
+  - You have [Terraform](https://registry.terraform.io/providers/scaleway/scaleway/latest/docs) installed on your machine
+
+</Message>
+
+## Context
+This tutorial will simulate a project with a production environment running all the time and a development environment that will be switched off on weekends to reduce costs.
+
+## Initialize a Terraform project
+
+1. Create a new folder, called `Terraform`, to store your configuration. Then enter the folder:
+
+    ```
+    mkdir Terraform && cd Terraform
+    ```
+
+2. Create five files to configure your infrastructure:
+    - `main.tf`: will contain the main set of configurations for your project. Here, it will be our Instance
+    - `provider.tf`: Terraform relies on plugins called “[providers](https://www.terraform.io/language/providers)” to interact with remote systems
+    - `backend.tf`: each Terraform configuration can specify a backend, which defines where the state file of the current infrastructure will be stored. Terraform uses this file to keep track of the managed resources. This state can be stored locally or remotely. Configuring a remote backend allows multiple people to work on the same infrastructure
+    - `variables.tf`: will contain the variable definitions for your project. Since all Terraform values must be defined, any variables that are not given a default value will become required arguments
+    - `terraform.tfvars`: allows you to set the actual value of the variables
+
+3. Create the following folders:
+    - `function`: to store your function code
+    - `files`: to temporary store your zip function code
+    Your folder structure should look like:
+    ```bash
+    Terraform
+    | -- files
+    | -- function
+    -- main.tf
+    -- backend.tf
+    -- provider.tf
+    -- terraform.tfvars
+    -- variables.tf
+    ```
+
+4. Edit the `backend.tf` file to enable remote configuration backup:
+    ```json
+        terraform {
+        backend "s3" {
+            bucket                      = "XXXXXXXXX"
+            key                         = "terraform.tfstate"
+            region                      = "fr-par"
+            endpoint                    = "https://s3.fr-par.scw.cloud"
+            skip_credentials_validation = true
+            skip_region_validation      = true
+        }
+        }
+        /*
+        For the credentials part:
+        ==> Create a ~/.aws/credentials:
```

Comment:
in tf doc I see there is a different path : https://registry.terraform.io/providers/scaleway/scaleway/latest/docs#shared-configuration-file 🤔

Reply:
that was the code from @julesmartin76000 tutorial 🧐

Reply:
This defines the S3 credentials, we define the SCW credentials as variables a bit below, no need to add a Scaleway configuration file for the tutorial normally we can use 
```
export AWS_ACCESS_KEY_ID=$SCW_ACCESS_KEY
export AWS_SECRET_ACCESS_KEY=$SCW_SECRET_KEY
``` 
in case we don't want to use the AWS config file